SwrveSDKCommon/Common/SwrveRESTClient.m

```diff
@@ -0,0 +1,119 @@
+#import "SwrveRESTClient.h"
+#import "SwrveCommonConnectionDelegate.h"
+#import "SwrveCommon.h"
+#import "SwrveAssetsManager.h"
+
+@interface SwrveConnectionDelegate : SwrveCommonConnectionDelegate
+
+@property(nonatomic, strong) SwrveRESTClient *restClient;
+
+- (id)init:(SwrveRESTClient *)restClient completionHandler:(ConnectionCompletionHandler)handler;
+
+@end
+
+@implementation SwrveConnectionDelegate
+
+@synthesize restClient;
+
+- (id)init:(SwrveRESTClient *)_restClient completionHandler:(ConnectionCompletionHandler)_handler {
+    self = [super init:_handler];
+    if (self) {
+        [self setRestClient:_restClient];
+    }
+    return self;
+}
+
+// Override default implementation and store the metricsString
+- (void)addHttpPerformanceMetrics:(NSString *)metricsString {
+    if (restClient) {
+        [restClient addHttpPerformanceMetrics:metricsString];
+    }
+}
+
+@end
+
+@implementation SwrveRESTClient
+
+@synthesize httpPerformanceMetrics;
+@synthesize timeoutInterval;
+
+- (id)initWithTimeoutInterval:(NSTimeInterval)timeoutInt {
+    self = [super init];
+    if (self) {
+        [self setTimeoutInterval:timeoutInt];
+    }
+    return self;
+}
+
+- (void)sendHttpGETRequest:(NSURL *)url queryString:(NSString *)query {
+    [self sendHttpGETRequest:url queryString:query completionHandler:nil];
+}
+
+- (void)sendHttpGETRequest:(NSURL *)url {
+    [self sendHttpGETRequest:url completionHandler:nil];
+}
+
+- (void)sendHttpGETRequest:(NSURL *)baseUrl queryString:(NSString *)query completionHandler:(void (^)(NSURLResponse *, NSData *, NSError *))handler {
+    NSURL *url = [NSURL URLWithString:query relativeToURL:baseUrl];
+    [self sendHttpGETRequest:url completionHandler:handler];
+}
+
+- (void)sendHttpGETRequest:(NSURL *)url completionHandler:(void (^)(NSURLResponse *, NSData *, NSError *))handler {
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url cachePolicy:NSURLRequestUseProtocolCachePolicy timeoutInterval:timeoutInterval]; // TODO self.config.httpTimeoutSeconds
```

Comment:
This breaks a current feature, it should be possible to pass self.config.httpTimeOutSeconds to this methods